app/src/main/java/com/duckduckgo/app/global/view/DaxDialog.kt

```diff
@@ -22,30 +22,44 @@ import android.graphics.Color
 import android.graphics.Rect
 import android.graphics.drawable.ColorDrawable
 import android.os.Bundle
+import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.fragment.app.DialogFragment
-import com.duckduckgo.app.browser.R
-import android.view.Gravity
-import kotlinx.android.synthetic.main.content_dax_dialog.*
 import android.view.animation.Animation
 import android.view.animation.OvershootInterpolator
 import android.view.animation.ScaleAnimation
 import android.widget.ImageView
 import android.widget.RelativeLayout
 import androidx.core.content.ContextCompat.getColor
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.browser.R
+import kotlinx.android.synthetic.main.content_dax_dialog.*
 
-class DaxDialog(
-    var daxText: String,
-    var buttonText: String,
+interface DaxDialog {
+    fun setDaxText(daxText: String)
+    fun setButtonText(buttonText: String)
+    fun setDialogAndStartAnimation()
+    fun onAnimationFinishedListener(onAnimationFinished: () -> Unit)
```

Comment:
The `onAnimationFinishedListener` is specific to the typing animation and is triggered when it finishes. I would suggest to move it to another interface or make it specific to the `TypewriteDaxDialog` class but this would make `DaxDialogHighlightView` a bit messier.
The other option is to assume the DaxDialog always is going to have the typing animation (which it does for now)  and maybe rename `TypewriteDaxDialog` to something more generic? Having `TypewriteDaxDialog` makes me think there is another version without the typing animation when in reality there isn't.
I don't think is massively important to stop the PR but it feels a very specific name for something that's always used like that.

Reply:
Actually, `onAnimationFinishedListener` can be moved into an interface and `DaxDialogHighlightView`will not be affected. The only part that will be affected is where we create those dialogs. But I liked that you commented on that. I think we can delay that change till we introduce a different Dax dialog, that is also the reason why I didn't segregate that interface into different ones.
As per the name, I'm fine changing the name or using this concrete one (because the interface takes the generic name), so I'm open to suggestions here. Tbh, I could come up with something else.